File: backend/src/main/java/com/andrew/hdss/dtos/FormDto.java
package com.andrew.hdss.dtos;

import com.andrew.hdss.models.Form;
import com.andrew.hdss.models.enums.FormCategory;
import com.andrew.hdss.models.enums.FormStatus;
import com.andrew.hdss.models.enums.FormTarget;

public record FormDto(
        Long id,
        String name,
        String title,
        FormCategory category,
        FormTarget target,
        Integer version,
        String description,
        boolean active,
        FormStatus status
) {
    public static FormDto from(Form form) {
        return new FormDto(
                form.getId(),
                form.getName(),
                form.getTitle(),
                form.getCategory(),
                form.getTarget(),
                form.getVersion(),
                form.getDescription(),
                form.isActive(),
                form.getStatus()
        );
    }
}

File: frontend/src/models/FormDto.ts
import type { FormCategory, FormStatus, FormTarget } from "./types/FormTypes";

export interface FormDto{
    id: number;
    name: string;
    title: string;
    category: FormCategory;
    target: FormTarget;
    version: number;
    description: string;
    active: boolean;
    status: FormStatus;
}